src/openrct2/ride/Ride.cpp

```diff
@@ -4579,6 +4579,41 @@ uint8_t ride_get_helix_sections(Ride* ride)
     return ride->special_track_elements & 0x1F;
 }
 
+bool Ride::HasElement(uint16_t element, bool useName) const
+{
+    bool moveSlowIt = true;
+    track_circuit_iterator it, slowIt;
+    int n = 0;
+    auto el = map_get_nth_element_at(stations[0].Start, n);
+    while (el == nullptr || el->GetType() != TileElementType::Track || el->AsTrack()->GetRideIndex() != id)
```

Comment:
This code is incorrect use the helper methods.

Reply:
Which helper methods?

Reply:
There are so many I couldn't quite be sure for this exact instance map_get_track_element_xxxx

Reply:
As far as I can tell all those only get a track element at a certain tile, which is not what this code does. This code looks whether a ride contains a certain track element.